src/controllers/controllerpresetinfo.h

```diff
@@ -48,7 +49,13 @@ class PresetInfo {
 
     inline const QString getPath() const { return m_path; }
 
-    inline const QString getName() const { return m_name; }
+    inline const QString getName() const {
+        if (m_name.length() != 0) {
+            return m_name;
+        } else {
+            return QFileInfo(m_path).baseName();
```

Comment:
When does this happen? Please comment.

Reply:
QFileInfo is quite expensive but includes caching. So a temporary QFileInfo school be avoided.

Reply:
This is called from enumeratePresets, which is called from DlgPrefController::slotUpdate

Reply:
You should initalize n_name in the constructor. 
You should make sure that only one QFileInfo is used.
This can become a member variable if required, or a local in the Constructor. 

This code place is also redundant now: 
https://github.com/mixxxdj/mixxx/blob/4f1e0832c0a8860efe3d5ab9e4725bebe023e3cc/src/controllers/dlgprefcontroller.cpp#L29